grafana/resource_organization.go

```diff
@@ -0,0 +1,305 @@
+package grafana
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	gapi "github.com/nytm/go-grafana-api"
+	"log"
+	"strconv"
+	"strings"
+)
+
+type OrgUser struct {
+	Id    int64
+	Email string
+	Role  string
+}
+
+const UserAdd = 10
+const UserUpdate = 20
+const UserRemove = 30
+
+type UserChange struct {
+	Type int8
+	User OrgUser
+}
+
+func ResourceOrganization() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateOrganization,
+		Read:   ReadOrganization,
+		Update: UpdateOrganization,
+		Delete: DeleteOrganization,
+		Exists: ExistsOrganization,
+		Importer: &schema.ResourceImporter{
+			State: ImportOrganization,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"admin_user": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "admin",
+			},
+			"create_users": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"org_id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"admins": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"editors": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"viewers": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func CreateOrganization(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gapi.Client)
+	name := d.Get("name").(string)
+	orgId, err := client.NewOrg(name)
+	if err != nil && err.Error() == "409 Conflict" {
+		return errors.New(fmt.Sprintf("Error: A Grafana Organization with the name '%s' already exists.", name))
+	}
+	if err != nil {
+		log.Printf("[DEBUG] creating Grafana organization %s", name)
```

Comment:
Are these `log.Printf` statements new? In the existing code, seems like it is only printed on errors?

Reply:
You're right - It looks like I used a lot more logging than the rest of the provider does. I'll cut that down to just the warning for removing an organization from the state like the other resources do. Thanks!